tests/test_unit/test_annotations/test_io.py

```diff
@@ -0,0 +1,524 @@
+from collections.abc import Callable
+from contextlib import nullcontext as does_not_raise
+from pathlib import Path
+from typing import Literal
+from unittest.mock import patch
+
+import pandas as pd
+import pytest
+
+from ethology.annotations.io import (
+    STANDARD_BBOXES_DF_COLUMNS,
+    STANDARD_BBOXES_DF_INDEX,
+    _df_bboxes_from_multiple_files,
+    _df_bboxes_from_single_file,
+    _df_bboxes_from_single_specific_file,
+    _df_rows_from_valid_COCO_file,
+    _df_rows_from_valid_VIA_file,
+    df_bboxes_from_files,
+)
+from ethology.annotations.validators import ValidCOCO, ValidVIA
+
+
+@pytest.fixture
+def multiple_input_files(annotations_test_data: dict) -> dict:
+    """Fixture that returns for each format, a pair of annotation files
+    with their number of annotations and images.
+    """
+    return {
+        "VIA": [
+            {
+                "path": annotations_test_data["VIA_JSON_sample_1.json"],
+                "n_annotations": 4440,
+                "n_images": 50,
+            },
+            {
+                "path": annotations_test_data["VIA_JSON_sample_2.json"],
+                "n_annotations": 3977,
+                "n_images": 50,
+            },
+        ],
+        "COCO": [
+            {
+                "path": annotations_test_data["COCO_JSON_sample_1.json"],
+                "n_annotations": 4344,
+                "n_images": 100,
+            },
+            {
+                "path": annotations_test_data["COCO_JSON_sample_2.json"],
+                "n_annotations": 4618,
+                "n_images": 100,
```

Comment:
is there any way you could auto-compute the n_annotations and n_images from each file (in a way that's orthogonal to the loaders you are testing)? As things currently are, you'll need to manually update these numbers if the file changes.
Moreover these hard-coded numbers re-appear in the parametrisation of several tests (such as `test_df_bboxes_from_single_specific_file`), which means they would have to be updated in multiple places.

Reply:
To re-formulate my points, there are two problems with this:
1. Even though you are defining the expected n annotations and images per file here, you are not consistently re-using that information in all tests, instead you supply the same values again via parametrisation.
2. The hard-coding of said values in the first place.

I guess I'm fine with 2 (if it's too much of a hustle to extract these values in an orthogonal way), but 1 should be fix-able?

Reply:
Both done!

It simplified the tests quite a bit, thanks Niko!